arduino_main/arduino_main.ino

```diff
@@ -0,0 +1,103 @@
+/* Copyright (c) 2019  Guyllaume Rousseau
+ * 
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ * 
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ * 
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+ 
+#include <ESP8266WiFi.h>
+#include <ESP8266HTTPClient.h>
+
+/*
+ * File confidential.h contains the ssid and password required to
+ * connect to a wifi network. They are kept seperate from the source
+ * code as they must stay private. DO NOT PUSH THEM TO GITHUB.
+ */
+#include "confidential.h"
+
+#define LIGHT_SIGNAL_INPUT 5 // pin D1
+
+int previousLightStatus = -1;
+HTTPClient client;
+
+void connectToWifi(char* ssid, char* password)
+{
+  WiFi.mode(WIFI_STA);
+  WiFi.begin(ssid, password);
+  int i = 0;
+  while(!isConnectedToWiFi())
+  {
+    delay(1000);
+    i++;
+    if(i>=30) // Fails after thirty seconds.
+    {
+      return;
+    }
+  }
+}
+
+void flashForError()
+{
+  for(int i=0; i<19; i++)
```

Comment:
Style: espaces entre les opérateurs comme `=` ou `<`

Reply:
19 itérations? pourquoi pas 20? Ça pourrait être une constante.

Reply:
J'aime les nombres premiers. Au début, j'avais pensé à dix-sept, mais j'ai choisi dix-neuf parce que nous sommes en 2019.

Reply:
Dans la première ligne d'une bouche for, je préfère ne pas écrire d'espaces. Ainsi, il est plus facile de distinguer ses trois éléments.

Reply:
C'est une façon de le voir.

Reply:
C'est pas moi qui décide si c'est acceptable, ça serrait plutôt Raphaël, mais dans la plupart des code source de d'autres projets, ça ne se ferait pas comme ça.

Reply:
J'essaie de laisser de côté mes préférences personnelles quand je code pour pas faire des trucs qui peuvent être un peu weird.